lib/internal/bootstrap/node.js

```diff
@@ -319,6 +319,27 @@ ObjectDefineProperty(process, 'features', {
 }
 
 const { emitWarning, emitWarningSync } = require('internal/process/warning');
+const { getOptionValue } = require('internal/options');
+
+let kTypeStrippingMode = null;
+ObjectDefineProperty(process.features, 'typescript', {
+  __proto__: null,
+  get() {
```

Comment:
This needs to emit experimental warning.

Reply:
I disagree, we should not discourage feature detection, that'd just nudge folks into either disabling warnings and/or find another way for detecting typescript support.

Also, I don't think this should be a getter, there's nothing dynamic about it.

Reply:
If we don't emit experimental warning here, and mark this as experimental, how can we remove this API if the `--experimental-typescript-*` flags are removed?

Reply:
Anything documented as experimental can be deleted at any time – and folks should write their code in a way that it still works whether `process.features.typescript` is `false` and `undefined` (they have to if they want their code to work on older release lines anyway), so removing it is unlikely to cause breakage.

I expect lots of folks to prefer not use an API that may emit a warning (because users don't like them), so they would have to find _another way_ to detect "TS" support, so what's the point of adding this in the first place if we don't want folks to use it?

Reply:
I agree, if typescript is enabled, it will emit the warning regardless, if typescript is disabled, it returns false. Even if it does not emit the warning we can remove it.

Reply:
Got it. I'll remove the emitted warning when I get a chance.

Reply:
Please don't resolve a review that isn't addressed. @RedYetiDev. 

I recommend looking into this once again @marco-ippolito @aduh95. If you insist, I'll remove my block but this is an experimental feature that we should emit experimental warning for (according to documentation)

Reply:
My apologies, I assumed this was resolved given the opposing opinions

Reply:
I insist, I’m -1 on landing this with a runtime warning. I’d recommend using stability 1 Experimental and not stability 1.0 nonsense if that’s where the point of contention lies.

Reply:
I marked it with the same stability that type strip support is labeled under, but I can change it